cmd/kerberos.go

```diff
@@ -0,0 +1,236 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+	"sync"
+
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	toolsWatch "k8s.io/client-go/tools/watch"
+	"k8s.io/klog"
+)
+
+type KerberosConfig struct {
+	NetPolIPs      []string
+	SidecarConfigs string
+}
+
+var kerberosCmd = &cobra.Command{
+	Use:   "kerberos",
+	Short: "Configure kerberos sidecar resources",
+	Long:  "Configure kerberos sidecar resources",
+	Run: func(cmd *cobra.Command, args []string) {
+		var wg sync.WaitGroup
+		// Create Kubernetes config
+		cfg, err := clientcmd.BuildConfigFromFlags(apiserver, kubeconfig)
+		if err != nil {
+			klog.Fatalf("error building kubeconfig: %v", err)
+		}
+
+		kubeClient, err := kubernetes.NewForConfig(cfg)
+		if err != nil {
+			klog.Fatalf("Error building kubernetes clientset: %s", err.Error())
+		}
+
+		kerberosConfig, err := getKerberosConfigmap(kubeClient)
+		if err != nil {
+			klog.Fatalf("Error getting configmap: %s", err.Error())
+		}
+
+		watchFunc := func(options metav1.ListOptions) (watch.Interface, error) {
+			timeOut := int64(60)
+			// Watches all namespaces, hence the Secrets("")
+			// Watches for all secrets named "kerberos-keytab"
+			return kubeClient.CoreV1().Secrets("").Watch(context.Background(), metav1.ListOptions{TimeoutSeconds: &timeOut,
+				FieldSelector: "metadata.name=kerberos-keytab"})
+		}
+		watcher, _ := toolsWatch.NewRetryWatcher("1", &cache.ListWatch{WatchFunc: watchFunc})
+		for event := range watcher.ResultChan() {
+			secret := event.Object.(*corev1.Secret)
+			switch event.Type {
+			case watch.Modified, watch.Added:
+				err := createKerberosUserConfigMap(secret.Namespace, kubeClient, kerberosConfig.SidecarConfigs)
+				if err != nil {
+					klog.Errorf("Error occurred while creating the ConfigMap for namespace %s: %s", secret.Namespace, err.Error())
+				}
+
+				err = createKerberosNetworkPolicy(secret.Namespace, kubeClient, kerberosConfig.NetPolIPs)
+				if err != nil {
+					klog.Errorf("Error occurred while creating the NetworkPolicy for namespace %s: %s", secret.Namespace, err.Error())
+				}
+			case watch.Error:
+				klog.Errorf("Kerberos secret in namespace %s contains an error.", secret.Namespace)
+			}
+		}
+
+		wg.Add(1)
+		wg.Wait()
+	},
+}
+
+func getKerberosConfigmap(client *kubernetes.Clientset) (KerberosConfig, error) {
```

Comment:
Let's get some function block comments too, even if the fns are straight forwards :)